src/web/lib/messagePicker.ts

```diff
@@ -90,39 +99,43 @@ export const pickMessage = ({
 		const candidateConfigsWithTimeout = candidates.map((c) =>
 			timeoutify(c, name),
 		);
-		const results = candidateConfigsWithTimeout.map((c) =>
-			c.candidate.canShow(),
+		const results: PendingMessage<any>[] = candidateConfigsWithTimeout.map(
+			(candidateConfig) => ({
+				candidateConfig,
+				canShow: candidateConfig.candidate.canShow(),
+			}),
 		);
 
-		const winner: Promise<WinningMessage> = results.reduce(
-			async (winningMessageSoFar, canShow, idx) => {
-				const runningIdx = (await winningMessageSoFar).idx;
-				if (runningIdx >= 0) {
-					return winningMessageSoFar;
-				}
+		const winnerResult = results.reduce<
+			Promise<WinningMessage<any> | null>
+		>(async (winningMessageSoFar, { candidateConfig, canShow }) => {
+			const running = await winningMessageSoFar;
```

Comment:
I wonder if there's a better name here than `running` 🤔  I know this is a hangover from my previous questionable choice of `runningIdx`, which was supposed to be like "running total". It's tricky because really `running` is the awaited/unwrapped `winningMessageSoFar`. I wonder if `winningMessageSoFar` should be `asyncWinningMessageSoFar` and then running could be just `winningMessageSoFar`? What do you reckon?

Reply:
Good point, and actually, since we don't use `running` later on this can just be:
`if (await winningMessageSoFar)`

Reply:
Ah great, that's a nice solution.